### PR TITLE
Create a script to generate place summaries.

### PR DESCRIPTION
**Pull request by keyurva**
* This is an initial script. @rvguha wants us to iterate on this before incorporating it into the website.

### CODE REVIEW

File: tools/summaries/main.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import json
+import logging
+
+from dc import get_ranking_csv
+from llm import get_summary
+
+logging.getLogger().setLevel(logging.INFO)
+
+_PLACE_JSON_FILE = "places.json"
+_SUMMARIES_JSON_FILE = "summaries.json"
```
Comment (keyurva) [inline tools/summaries/main.py:25]: @dwnoble - let me know where you want the json file to go.

File: tools/summaries/dc.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# limitations under the License.
+
+import json
+import logging
+
+import pandas as pd
+import requests
+
+_API_BASE_URL = "https://browser.datacommons.org/api/place/ranking"
+_RANKINGS_HEADER = "Rankings (in)"
```
Comment (beets) [inline tools/summaries/dc.py:22]: we could drop the parenthesis in the header. i wonder if that would make the csv easier to interpret by palm

File: tools/summaries/main.py (hunk 1 above)
Comment (beets) [reply tools/summaries/main.py:25]: this should go in a new folder in the [`config dir`](https://github.com/datacommonsorg/website/tree/master/server/config)

File: tools/summaries/palm.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import re
+
+import dc
+import requests
+
+_API_KEY = os.getenv("PALM_API_KEY")
+_API_URL = "https://generativelanguage.googleapis.com/v1beta2/models/text-bison-001:generateText"
+_TEMPERATURE = 0.2
+# Sometimes the summaries may not be factual. Retry them in that case.
+# We use a heuristic that the summary must contain at least 2 numbers to be factual.
```
Comment (beets) [inline tools/summaries/palm.py:27]: we could also verify that the returned numbers matches a value in the prompt csv.

File: tools/summaries/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Generate Place Summaries
+
+This script generates NL summaries for places.
+
```
Comment (beets) [inline tools/summaries/README.md:4]: Should also specify here the environment values that the scripts rely on (and assert if it's unavailable).

**Review by beets — APPROVED** [review]
thanks keyur! and sorry for the late review.

even though we're experimenting, would be good to have more documentation as we iterate since we are not in crunch mode anymore (eg module comments, etc).

File: tools/summaries/main.py (hunk 1 above)
Comment (keyurva) [reply tools/summaries/main.py]: Thanks. Done.

File: tools/summaries/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Generate Place Summaries
+
+This script generates NL summaries for places.
```
[the remark this answers is not in the document]
Comment (keyurva) [reply tools/summaries/README.md]: Done.

File: tools/summaries/dc.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+
+import pandas as pd
+import requests
+
+_API_BASE_URL = "https://browser.datacommons.org/api/place/ranking"
```
[the remark this answers is not in the document]
Comment (keyurva) [reply tools/summaries/dc.py]: Done.

File: tools/summaries/dc.py (hunk 2 above)
Comment (keyurva) [reply tools/summaries/dc.py]: Done. Summaries seem to be similar with this change. In the larger context, these are the types of things we'll need to tinker with to come up with the prompt(s) that works best.

File: tools/summaries/openai.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import json
+import logging
+import os
+
+import requests
+
+_API_KEY = os.getenv("OPENAI_API_KEY")
+_API_URL = "https://api.openai.com/v1/chat/completions"
+_MODEL_NAME = "gpt-3.5-turbo"
+_TEMPERATURE = 0.5
```
[the remark this answers is not in the document]
Comment (keyurva) [reply tools/summaries/openai.py:26]: Done.

File: tools/summaries/README.md (hunk 4 above)
Comment (keyurva) [reply tools/summaries/README.md:4]: Good idea. Done.

**Comment by keyurva** [comment]
Thanks Carolyn.

**Pull request merged**